src/utils.js

```diff
@@ -0,0 +1,112 @@
+'use strict'
+
+const async = require('async')
+const fs = require('fs')
+const hat = require('hat')
+const os = require('os')
+const path = require('path')
+const exec = require('./exec')
+
+const join = path.join
+
+const isWindows = os.platform() === 'win32'
+
+// taken from https://github.com/hughsk/flat
+exports.flatten = (target) => {
+  let output = {}
+  const step = (object, prev) => {
+    object = object || {}
+    Object.keys(object).forEach(function (key) {
+      var value = object[key]
+      var isarray = Array.isArray(value)
+      var type = Object.prototype.toString.call(value)
+      var isbuffer = Buffer.isBuffer(value)
+      var isobject = (
+        type === '[object Object]' ||
+        type === '[object Array]'
+      )
+
+      var newKey = prev
+        ? prev + '.' + key
+        : key
+
+      if (!isarray && !isbuffer && isobject && Object.keys(value).length) {
+        return step(value, newKey)
+      }
+
+      output[newKey] = value
+    })
+  }
+
+  step(target)
+
+  return output
+}
+
+function tryJsonParse (input, callback) {
```

Comment:
if you really need to do the JSON parse, may I suggest [safe-json-parse](https://www.npmjs.com/package/safe-json-parse)?